llvm/lib/Target/AMDGPU/AMDGPULateCodeGenPrepare.cpp

```diff
@@ -296,9 +296,8 @@ bool LiveRegOptimizer::optimizeLiveType(
       // Collect all uses of PHINodes and any use the crosses BB boundaries.
       if (UseInst->getParent() != II->getParent() || isa<PHINode>(II)) {
         Uses.insert(UseInst);
-        if (!Defs.count(II) && !isa<PHINode>(II)) {
```

Comment:
Why is this redundant?

Reply:
`Defs.count(II)` is redundant because `Defs.insert(II)` checks the membership before inserting `II` anyway.